src/main/java/com/metamx/common/parsers/JavascriptParser.java

```diff
@@ -0,0 +1,154 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.metamx.common.parsers;
+
+import com.google.common.base.Function;
+import com.google.common.base.Optional;
+import com.google.common.base.Splitter;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import com.metamx.common.collect.Utils;
+import com.metamx.common.logger.Logger;
+import org.mozilla.javascript.Context;
+import org.mozilla.javascript.ContextFactory;
+import org.mozilla.javascript.ScriptableObject;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+/**
+ */
+public class JavascriptParser implements Parser<String, Object>
+{
+  private static final Logger log = new Logger(JavascriptParser.class);
+
+  private static Function<Object, Object> compile(String function)
+  {
+    final ContextFactory contextFactory = ContextFactory.getGlobal();
+    final Context context = contextFactory.enterContext();
+    context.setOptimizationLevel(9);
+
+    final ScriptableObject scope = context.initStandardObjects();
+
+    final org.mozilla.javascript.Function fn = context.compileFunction(scope, function, "fn", 1, null);
+    Context.exit();
+
+    return new Function<Object, Object>()
+    {
+      public Object apply(Object input)
+      {
+        // ideally we need a close() function to discard the context once it is not used anymore
+        Context cx = Context.getCurrentContext();
+        if (cx == null) {
+          cx = contextFactory.enterContext();
+        }
+
+        final Object res = fn.call(cx, scope, scope, new Object[]{input});
+        return res != null ? Context.toObject(res, scope) : null;
+      }
+    };
+  }
+
+  private final Function<Object, Object> fn;
+  private final String listDelimiter;
+  private final Splitter listSplitter;
+  private final Function<String, Object> valueFunction;
+
+  private ArrayList<String> fieldNames = null;
+
+  public JavascriptParser(
+      final String function,
+      final Optional<String> listDelimiter
+  )
+  {
+    this.fn = compile(function);
+    this.listDelimiter = listDelimiter.isPresent() ? listDelimiter.get() : Parsers.DEFAULT_LIST_DELIMITER;
+    this.listSplitter = Splitter.on(this.listDelimiter);
+    this.valueFunction = new Function<String, Object>()
+    {
+      @Override
+      public Object apply(String input)
+      {
+        if (input.contains(JavascriptParser.this.listDelimiter)) {
+          return Lists.newArrayList(
+              Iterables.transform(
+                  listSplitter.split(input),
+                  ParserUtils.nullEmptyStringFunction
+              )
+          );
+        } else {
+          return ParserUtils.nullEmptyStringFunction.apply(input);
+        }
+      }
+    };
+  }
+
+  public JavascriptParser(
```

Comment:
we're not 100% consistent, but we usually call things `JavaScript` rather than `Javascript` so I would go with the former.